slidechain/postexport.go

```diff
@@ -0,0 +1,167 @@
+package slidechain
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"math"
+	"time"
+
+	"github.com/chain/txvm/crypto/ed25519"
+	i10rjson "github.com/chain/txvm/encoding/json"
+	"github.com/chain/txvm/errors"
+	"github.com/chain/txvm/protocol/bc"
+	"github.com/chain/txvm/protocol/txvm"
+	"github.com/chain/txvm/protocol/txvm/op"
+	"github.com/chain/txvm/protocol/txvm/txvmutil"
+	"github.com/stellar/go/xdr"
+)
+
+func (c *Custodian) doPostExport(ctx context.Context, assetXDR, anchor, txid []byte, amount, seqnum, peggedOut int64, exporter, temp string, pubkey []byte) error {
+	var asset xdr.Asset
+	err := asset.UnmarshalBinary(assetXDR)
+	if err != nil {
+		return errors.Wrap(err, "unmarshaling asset xdr")
+	}
+	assetBytes, err := asset.MarshalBinary()
+	if err != nil {
+		return errors.Wrap(err, "marshaling asset bytes")
+	}
+	assetID := bc.NewHash(txvm.AssetID(importIssuanceSeed[:], assetBytes))
+	ref := struct {
```

Comment:
This struct definition should not be repeated in multiple places, which could drift out of sync with each other. It should have its own `type`.